rma_account/models/procurement.py

```diff
@@ -33,8 +33,12 @@ def _get_stock_move_values(
             line = self.env["rma.order.line"].browse([line])
             move = line.reference_move_id
             if move and move.stock_valuation_layer_ids:
-                cost = move.stock_valuation_layer_ids[-1].unit_cost
-                res["price_unit"] = cost
+                layers = move.stock_valuation_layer_ids
+                price_unit = sum(layers.mapped("value")) / sum(
+                    layers.mapped("quantity")
+                )
+
+                res["price_unit"] = price_unit
```

Comment:
does this makes sense for FIFO?

Reply:
@LoisRForgeFlow Makes sense, when you apply landed costs because one move will have the main layer and then subsequent layers with extra cost due to landed costs.